src/main/java/com/iridium/iridiumskyblock/listeners/VehicleDamageListener.java

```diff
@@ -0,0 +1,29 @@
+package com.iridium.iridiumskyblock.listeners;
+
+import com.iridium.iridiumcore.utils.StringUtils;
+import com.iridium.iridiumskyblock.IridiumSkyblock;
+import com.iridium.iridiumskyblock.PermissionType;
+import com.iridium.iridiumskyblock.database.Island;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.vehicle.VehicleDamageEvent;
+
+import java.util.Optional;
+
+public class VehicleDamageListener implements Listener {
+
+    @EventHandler
+    public void onVehicleDamage(VehicleDamageEvent event) {
+        if (event.getAttacker() != null && event.getAttacker() instanceof Player) {
+            Player attacker = (Player) event.getAttacker();
+            Optional<Island> island = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getVehicle().getLocation());
+            if (island.isPresent()) {
+                if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), IridiumSkyblock.getInstance().getUserManager().getUser(attacker), PermissionType.BLOCK_BREAK)) {
```

Comment:
Boats and carts isn't a block too (like 🐴  ), so permission type must be KILL_MOBS

Reply:
Minecarts are no Mobs

Reply:
Check the Superinterfaces [here](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Mob.html)

Reply:
no, i mean entity sorry

Reply:
We don't have KILL_ENTITIES, but this would indeed be a better alternative
```suggestion
                if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), IridiumSkyblock.getInstance().getUserManager().getUser(attacker), PermissionType.INTERACT_ENTITIES)) {
```

Reply:
Me and Nubebuster talked about this in pm's we deided on making a new category